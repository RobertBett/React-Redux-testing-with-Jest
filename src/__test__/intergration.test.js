import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import moxios from 'moxios';
import Root from '../Root';
import App from '../components/App';


describe('Intergration Tests', () => {
    let wrapped;
    beforeEach(() =>{
        moxios.install();
        moxios.stubRequest('http://jsonplaceholder.typicode.com/comments',{
            status: 200,
            response: [{name: 'fetched 1'}, { name:'whatever'}]
        })

        wrapped = mount(
            <Root>
                <App/>
            </Root>
        );
    });
    afterEach(() =>{
        moxios.uninstall()
    });
    
    it('can fetch a list of comments and display them', (done) =>{
        // Attemp to render the entire app
        // find the fetchComments button and click it 
        wrapped.find('#fetch-button').simulate('click');
        //expect to find a list of comments 
        setTimeout(() => {
            wrapped.update()
            expect(wrapped.find('li').length).toEqual(2);   
            done()
        }, 1000);
    })
    it('should match the snapshot', () =>{
        expect(toJSON(wrapped)).toMatchSnapshot()
    })
})