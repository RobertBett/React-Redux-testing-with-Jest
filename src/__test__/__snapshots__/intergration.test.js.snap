// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intergration Tests should match the snapshot 1`] = `
<Component>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Component>
      <div
        style={
          Object {
            "margin": "50px",
          }
        }
      >
        <Connect(CommentBox)>
          <CommentBox
            fetchComments={[Function]}
            saveComment={[Function]}
          >
            <div>
              <form
                onSubmit={[Function]}
              >
                <h4>
                  Add a Comment 
                </h4>
                <textarea
                  cols="30"
                  id=""
                  onChange={[Function]}
                  rows="10"
                  value=""
                />
                <div>
                  <button>
                    Submit Comment
                  </button>
                </div>
              </form>
              <button
                id="fetch-button"
                onClick={[Function]}
              >
                Fetch Comments
              </button>
            </div>
          </CommentBox>
        </Connect(CommentBox)>
        <Connect(CommentList)>
          <CommentList
            comments={Array []}
            dispatch={[Function]}
          >
            <div>
              <ul />
            </div>
          </CommentList>
        </Connect(CommentList)>
      </div>
    </Component>
  </Provider>
</Component>
`;
